src/Connector/ArrayConverter/FlatToStandard/CategoryDecorator.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flagbit\Bundle\CategoryBundle\Connector\ArrayConverter\FlatToStandard;
+
+use Akeneo\Tool\Component\Connector\ArrayConverter\ArrayConverterInterface;
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+use function explode;
+use function in_array;
+
+/**
+ * Decorator for the category flat to standard array converter that cares about category properties.
+ *
+ * This decorator extends the original category flat to standard array converter and ensures
+ * that the custom category property data is processed too. The processed properties
+ * are stored in a {@see ParameterBag} to allow retrieval in other places.
+ *
+ * The modifications done by this decorator do not affect the converted result
+ * of the original category flat to standard array converter in any way.
+ */
+class CategoryDecorator implements ArrayConverterInterface
+{
+    private ArrayConverterInterface $baseConverter;
+    /** @var ParameterBag<string, array<string, mixed>> */
+    private ParameterBag $propertiesBag;
+
+    /**
+     * @phpstan-param ParameterBag<string, array<string, mixed>> $propertiesBag
+     */
+    public function __construct(
+        ArrayConverterInterface $baseConverter,
+        ParameterBag $propertiesBag
+    ) {
+        $this->baseConverter = $baseConverter;
+        $this->propertiesBag = $propertiesBag;
+    }
+
+    /**
+     * @param array<string, mixed> $item
+     * @param array<string, mixed> $options
+     *
+     * @return array<string, mixed>
+     */
+    public function convert(array $item, array $options = []): array
+    {
+        $originalResult = $this->baseConverter->convert($item, $options);
+        if (! isset($originalResult['code'])) {
+            return $originalResult;
+        }
+
+        $this->propertiesBag->set($item['code'], $this->extractCategoryProperties($item));
+
+        return $originalResult;
+    }
+
+    /**
+     * Extract the custom category properties from the supplied item.
+     *
+     * The array of properties returned by this function already has the proper
+     * format of {@see CategoryProperty::$properties} for further processing.
+     *
+     * @param array<string, mixed> $item
+     *
+     * @return array<string, mixed>
+     */
+    private function extractCategoryProperties(array $item): array
+    {
+        $categoryProperties = [];
+        $defaultProperties  = ['code', 'label', 'parent'];
+
+        foreach ($item as $fieldName => $field) {
+            if (in_array(explode('-', $fieldName)[0], $defaultProperties, true)) {
```

Comment:
```suggestion
            if (in_array($this->getPropertyName($fieldName), $defaultProperties, true)) {
```

I think this should to the same. Also: Please rename it toi `getPropertyCode()`